gradle-conjure-api/src/main/java/com/palantir/gradle/conjure/api/GeneratorOptions.java

```diff
@@ -87,4 +87,28 @@ public GeneratorOptions addFlag(String flag) {
         this.setProperty(flag, true);
         return this;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        GeneratorOptions that = (GeneratorOptions) other;
+
+        return storage.equals(that.storage);
+    }
+
+    @Override
+    public int hashCode() {
+        return storage.hashCode();
+    }
```

Comment:
This class was used as a task input, I don't know how gradle calculates that sort of thing, but I suspect the presence of a reasonable equals/hashCode doesn't hurt. I couldn't find docs on how task inputs are compared, I'd really appreciate if someone could point me in the right direction!

Reply:
All inputs need to be serializable. Gradle then serializes those to a binary format and compares those binary formats to see if the input has changed (think it compares checksums of the serialized representation but not super sure about the details here).

So hashcode shouldn't be necessary but yeah, doesn't hurt to have one!

Reply:
Thanks!